v2/header.go

```diff
@@ -163,11 +163,38 @@ func insertMetadata(ctx context.Context, keyvals ...string) metadata.MD {
 		out = metadata.MD(make(map[string][]string))
 	}
 	headers := callctx.HeadersFromContext(ctx)
-	for k, v := range headers {
-		out[k] = append(out[k], v...)
+
+	// x-goog-api-client is a special case that we want to make sure gets merged
+	// into a single header.
+	const xGoogHeader = "x-goog-api-client"
+	var mergedXgoogHeader strings.Builder
+
+	for k, vals := range headers {
+		out[k] = append(out[k], vals...)
+
+		if k == xGoogHeader {
+			// Merge all values for the x-goog-api-client header set on the ctx.
+			for _, v := range vals {
+				mergedXgoogHeader.WriteString(v)
+				mergedXgoogHeader.WriteRune(' ')
+			}
+		}
 	}
 	for i := 0; i < len(keyvals); i = i + 2 {
 		out[keyvals[i]] = append(out[keyvals[i]], keyvals[i+1])
```

Comment:
Did you want to skip setting this for the x-goog-api-client header? Maybe move it after the `if` statement below, and add a `continue` statement to the end of the `if` block? This is not necessary due to the overwrite on line 196, but might seem cleaner.

Reply:
Makes sense, I'll change that.